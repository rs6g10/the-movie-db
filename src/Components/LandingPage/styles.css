.movie {
    margin: 15px;
}

.movies-grid {
    position: relative;
    display: grid;
    grid-template-columns: repeat(4,1fr);
    grid-gap: 3rem 2rem;
}

.movie-container {
    position: relative;
    overflow: hidden;
    height: 100%;
    max-height: 310px;
}
.movie-info {
    display: grid;
    justify-content: center;
    align-items: center;
    position: absolute;
    text-align: center;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0;
    transition: 0.2s opacity ease;
    will-change: opacity;
    z-index: 10;
}
.release_date {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 0.8rem;
    color: white;
    opacity: 0.7;
    padding: 0.8rem 0rem;
}
.movie_title {
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0rem 1rem;
    text-align: center;
    color: white;
    font-size: 1.1rem;
}

.vote {
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 1.4rem;
    padding: 0rem 1rem;
    color: #FFD700;
}

.movie-container:before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(2, 2, 2, 0.9);
    opacity: 0;
    transition: 0.2s opacity ease;
    will-change: opacity;
    z-index: 5;
}
.movie-container img {
    width: 100%;
    transition: 0.2s transform ease;
    will-change: transform;
}

.movie-container:hover {
    cursor: pointer;
}
.movie-container:hover .movie-info {
    opacity: 1;
}
.movie-container:hover:before {
    opacity: 1;
}
.movie-container:hover img {
    transform: scale(1.05);
}

.welcome-categories {
    display: grid;
    grid-template-columns: 33% 33% 33%;
    grid-template-areas: "now_playing upcoming popular";
    text-transform: capitalize;
    align-items: center;
    justify-content: center;
    color: white;
}

.welcome-category {
    transition: color 0.15s ease;
    padding: 1rem 0rem;
    text-align: center;
}

.welcome-category:hover {
    color: #01d277;
    cursor: pointer;
}

.now_playing {
    grid-area: now_playing;
}

.upcoming {
    grid-area: upcoming;
}

.popular {
    grid-area: popular;
}

.active {
    color: #01d277;
    font-weight: bold;
}

@media (max-width: 767px) {
    .movie-container {
        max-height: 108px;
        height: 108px;
    }

    .vote {
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: 0.8rem;
        color: #FFD700;
    }
}
