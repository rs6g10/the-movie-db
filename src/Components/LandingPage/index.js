import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classnames from 'classnames';
import { parseResponse } from '../../utility';

import './styles.css';

class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      error: null,
      results: null,
      selectedCategory: 'now_playing',
    };
    this.selectCategory = this.selectCategory.bind(this);
  }

  componentDidMount() {
    const url = `https://api.themoviedb.org/3/movie/now_playing?api_key=${this.props.apiKey}`;
    this.fetchApi(url);
  }

  selectCategory(key) {
    this.setState({ selectedCategory: key });
    const url = `https://api.themoviedb.org/3/movie/${key}?api_key=${this.props.apiKey}`;
    this.fetchApi(url);
  }

  fetchApi(url) {
    this.setState({ loading: true });
    fetch(url).then(res => parseResponse(res)).then((datum) => {
      if (this.refs.moviesGrid) {
        this.setState({ loading: false, error: null });
        if (datum.results.length > 0) {
          this.setState({
            results: datum.results,
          });
        }
      }
    }).catch((err) => {
      this.setState({
        error: err,
        loading: false,
      });
    });
  }

  render() {
    const selectedCategory = this.state.selectedCategory;
    return (
      <div>
        <div className="welcome-categories">
          {this.props.categories.map(category => (
            <div
              className={classnames(category.key, 'welcome-category',
                selectedCategory === category.key ? 'active' : '')}
              onClick={() => this.selectCategory(category.key)}
              aria-hidden
              key={`category-${category.key}`}
            >{category.title}</div>
          ))
          }
        </div>
        <div className="movies-grid" ref="moviesGrid">
          {this.state.loading &&
          <div className="row">
            <span className="loading">Loading...</span>
          </div>}
          {!this.state.loading && !this.state.error &&
          this.state.results && this.state.results.map(result => (
            <Link to={`/movie/${result.id}`} key={`link-${result.id}`}>
              <div
                className="movie-container relative"
                key={`container-${result.id}`}
              >
                <img
                  src={`https://image.tmdb.org/t/p/w500/${result.poster_path}`}
                  alt="movie-poster"
                />
                <div className="movie-info">
                  <div className="release_date">{result.release_date}</div>
                  <div className="movie_title">{result.title}</div>
                </div>
              </div>
              <div className="vote">
                <span role="img"
                      aria-label="star">‚≠ê</span> {result.vote_average}
              </div>
            </Link>
          ))}
          {this.state.error &&
          <div className="row">
            <span className="error">{this.state.error.message}</span>
          </div>}
        </div>
      </div>
    );
  }
}

LandingPage.propTypes = {
  categories: PropTypes.array,
  apiKey: PropTypes.string.isRequired,
};

LandingPage.defaultProps = {
  categories: [
    { title: 'Now Showing', key: 'now_playing' },
    { title: 'Upcoming', key: 'upcoming' },
    { title: 'Popular', key: 'popular' },
  ],
};

export default LandingPage;

