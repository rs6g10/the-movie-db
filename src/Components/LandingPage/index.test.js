import '../../test/setupEnzyme';

import React from 'react';
import { shallow } from 'enzyme';

import LandingPage from './';

describe('LandingPage', () => {
  it('should render as expected with correct state', () => {
    const props = { apiKey: 'blah' };
    const element = shallow(<LandingPage {...props} />);

    expect(element).toMatchSnapshot();
  });
});
