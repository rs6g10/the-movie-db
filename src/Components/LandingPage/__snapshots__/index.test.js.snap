// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render as expected with correct state 1`] = `
<div>
  <div
    className="welcome-categories"
  >
    <div
      aria-hidden={true}
      className="now_playing welcome-category active"
      key="category-now_playing"
      onClick={[Function]}
    >
      Now Showing
    </div>
    <div
      aria-hidden={true}
      className="upcoming welcome-category"
      key="category-upcoming"
      onClick={[Function]}
    >
      Upcoming
    </div>
    <div
      aria-hidden={true}
      className="popular welcome-category"
      key="category-popular"
      onClick={[Function]}
    >
      Popular
    </div>
  </div>
  <div
    className="movies-grid"
  >
    <div
      className="row"
    >
      <span
        className="loading"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
